streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -373,6 +393,25 @@ class StreamingContext private[streaming] (
   }
 
   /**
+   * Create an input stream that monitors a Hadoop-compatible filesystem
+   * for new files and reads them as flat binary files, assuming a fixed length per record,
+   * generating one byte array per record. Files must be written to the monitored directory
+   * by "moving" them from another location within the same file system. File names
+   * starting with . are ignored.
+   * @param directory HDFS directory to monitor for new file
+   * @param recordLength length of each record in bytes
+   */
+  def binaryRecordsStream(
+      directory: String,
+      recordLength: Int): DStream[Array[Byte]] = {
+    val conf = sc_.hadoopConfiguration
+    conf.setInt(FixedLengthBinaryInputFormat.RECORD_LENGTH_PROPERTY, recordLength)
+    val br = fileStream[LongWritable, BytesWritable, FixedLengthBinaryInputFormat](directory, conf)
+    val data = br.map{ case (k, v) => v.getBytes}
```

Comment:
Probably no need to make a copy if this is actually safe (it would just harm performance), so I think it would be fine to add a comment.  I suppose we could also do an assert to check that the array isn't padded, since that's unlikely to be expensive.